version: "3.9"
services:
  customers:
    container_name: api
    build:
      context: .
      dockerfile: Dockerfile
    working_dir: /opt/customers/
    command: bash -c "python manage.py runserver 0:80"
    env_file:
      - .env
    expose:
      - "80"
    networks:
      customers:
        ipv4_address: 172.16.239.10
    volumes:
      - ./customers:/opt/customers
      - ./customers/my.cnf:/etc/my.cnf
    depends_on:
      - mysql

  mysql:
    container_name: data_base
    image: "mysql:5.7"
    expose:
      - "3306"
      - "33060"
    command: ['--character-set-server=utf8mb4', '--collation-server=utf8mb4_unicode_ci', "--max-connections=500"]
    environment:
      MYSQL_ROOT_PASSWORD: secret
      MYSQL_DATABASE: customers
    networks:
      customers:
        ipv4_address: 172.16.239.11

networks:
  customers:
    driver: bridge
    ipam:
      driver: default
      config:
        - subnet: 172.16.239.0/24
          gateway: 172.16.239.1